olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -24,14 +24,64 @@
 #include <sstream>
 #include <thread>
 
+#include <olp/core/http/HttpResponse.h>
 #include <olp/core/network/HttpResponse.h>
 #include "NetworkAsyncHandlerImpl.h"
 #include "olp/core/network/Network.h"
 #include "olp/core/network/NetworkResponse.h"
+#include "olp/core/utils/Url.h"
 
 namespace olp {
 namespace client {
 
+
+namespace {
+
+CancellationToken ExecuteSingleRequest(
+    std::shared_ptr<http::Network> network,
+    const http::NetworkRequest& request,
+    const NetworkAsyncCallback& callback) {
+
+  auto response_body = std::make_shared<std::stringstream>();
+  auto send_outcome = network->Send(
+    request, response_body,
+    [network, callback, response_body](const http::NetworkResponse& response) {
+      // status is -1 if it's timed-out
+      network::HttpResponse result(response.GetStatus());
+
+      if (result.status >= 400 || result.status < 0) {
+        result.response = "Error occured. Please check HTTP status code. " + std::move(response_body->str());
+
+      } else {
+        result.response = std::move(response_body->str());
+      }
+
+      callback(result);
+    }
+  );
+
+  if (!send_outcome.IsSuccessfull()) {
+    network::HttpResponse result;
+    result.status = int(http::ErrorCode::IO_ERROR);
```

Comment:
```suggestion
    result.status = send_outcome.GetErrorCode();
```
What is the reason to use `IO_ERROR` here?